lib/sprockets/utils.rb

```diff
@@ -76,16 +76,15 @@ def hash_reassoc(hash, key_a, key_b = nil, &block)
     def string_end_with_semicolon?(str)
       i = str.size - 1
       while i >= 0
-        c = str[i]
+        c = str[i].ord
         i -= 1
-        if c == "\n" || c == " " || c == "\t"
-          next
-        elsif c != ";"
-          return false
-        else
-          return true
+
+        # Need to compare against the ordinals because the string can be UTF_8 or UTF_32LE encoded
+        unless c == 0x0A || c == 0x20 || c == 0x09
+          return c === 0x3B
```

Comment:
Might be worth mentioning what those ASCII codes stand for in comment (saves a table lookup for most people)